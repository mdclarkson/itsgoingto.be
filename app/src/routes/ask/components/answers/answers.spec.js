import React from 'react'
import {
  render, screen
} from '@testing-library/react'
import Answers from './answers'

const defaultProps = {
  hasQuestion: true,
  answers: [],
  onAnswerChange: jest.fn(),
  onRemoveAnswer: jest.fn()
}

describe('(Route) Ask', () => {
  describe('(Component) Answers', () => {
    describe('(Render)', () => {
      describe('when hasQuestion is true', () => {
        it('answers does not have class gone', () => {
          render(<Answers {...defaultProps} hasQuestion />)

          const options = screen.getByTestId('answers')
          expect(options).not.toHaveClass('gone')
        })
      })

      describe('when hasQuestion is false', () => {
        it('answers has class gone', () => {
          render(<Answers {...defaultProps} hasQuestion={false} />)

          const options = screen.getByTestId('answers')
          expect(options).toHaveClass('gone')
        })
      })

      describe('with answers as strings', () => {
        it('matches snapshot', () => {
          const answers = [
            'Answer 1',
            'Answer 2',
            'Answer 3'
          ]
          const { asFragment } = render(<Answers {...defaultProps} answers={answers} />)

          expect(asFragment()).toMatchSnapshot()
        })
      })

      describe('with answers as objects', () => {
        it('matches snapshot', () => {
          const answers = [
            { id: 4, answer: 'Answer 4' },
            { id: 5, answer: 'Answer 5' },
            { id: 6, answer: 'Answer 6' }
          ]
          const { asFragment } = render(<Answers {...defaultProps} answers={answers} />)

          expect(asFragment()).toMatchSnapshot()
        })
      })
    })
  })
})
