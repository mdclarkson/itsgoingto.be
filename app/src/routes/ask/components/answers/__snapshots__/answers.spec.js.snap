// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Component) Answers (Render) with answers as objects matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="answers hideable"
    data-testid="answers"
  >
    <label
      class="input-label input-label-answers"
    >
      Add some responses
    </label>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-0"
      >
        1
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-0"
        name="answer-0"
        type="text"
        value="Answer 4"
      />
    </div>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-1"
      >
        2
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-1"
        name="answer-1"
        type="text"
        value="Answer 5"
      />
    </div>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-2"
      >
        3
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-2"
        name="answer-2"
        type="text"
        value="Answer 6"
      />
    </div>
    <div
      class="input input-answer input-disabled"
    >
      <label
        class="input-label input-label-answer"
        for="answer-3"
      >
        4
      </label>
      <input
        class="input-field input-field-answer"
        disabled=""
        id="answer-3"
        name="answer-3"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) Ask (Component) Answers (Render) with answers as strings matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="answers hideable"
    data-testid="answers"
  >
    <label
      class="input-label input-label-answers"
    >
      Add some responses
    </label>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-0"
      >
        1
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-0"
        name="answer-0"
        type="text"
        value="Answer 1"
      />
    </div>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-1"
      >
        2
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-1"
        name="answer-1"
        type="text"
        value="Answer 2"
      />
    </div>
    <div
      class="input input-answer"
    >
      <label
        class="input-label input-label-answer"
        for="answer-2"
      >
        3
      </label>
      <input
        class="input-field input-field-answer"
        id="answer-2"
        name="answer-2"
        type="text"
        value="Answer 3"
      />
    </div>
    <div
      class="input input-answer input-disabled"
    >
      <label
        class="input-label input-label-answer"
        for="answer-3"
      >
        4
      </label>
      <input
        class="input-field input-field-answer"
        disabled=""
        id="answer-3"
        name="answer-3"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
